src/pieces.py

```diff
@@ -0,0 +1,9 @@
+from enum import Enum, auto
+
+class Pieces(Enum):
```

Comment:
I don't see that `Pieces` are ever enumerated.

Another issue is that you can't `from src.pieces import X, O`, atm you can only:
```python
from src.pieces import Pieces
X = Pieces.X
O = Pieces.Y  # Repeated everywhere.
```

Simples solution is `X = 'X'` or `X = Mark('X')` if you don't it to be mistaken and comparable to `'X'`.